test-monitor/nitestmonitor.yml

```diff
@@ -0,0 +1,1277 @@
+swagger: '2.0'
+info:
+  version: '1'
+  title: Test Monitor Web Service
+  description: Test Monitor HTTP API
+  contact:
+    name: National Instruments
+    url: 'https://www.ni.com/systemlink'
+    email: support@ni.com
+basePath: /nitestmonitor
+consumes:
+  - application/json
+produces:
+  - application/json
+securityDefinitions:
+  basicAuth:
+    type: basic
+  cookieAuth:
+    type: apiKey
+    in: header
+    name: Cookie
+security:
+  - basicAuth: []
+  - cookieAuth: []
+x-ni-routing-key: Skyline.TestMonitor
+definitions:
+  Error:
+    description: Contains error information
+    type: object
+    properties:
+      name:
+        description: The string error code
+        type: string
+        example: Skyline.OneOrMoreErrorsOccurred
+      code:
+        description: Numeric error code
+        type: integer
+        example: -251040
+      resourceType:
+        description: The type of resource associated with the error
+        type: string
+        example: TestResult
+      resourceId:
+        description: The identifier of the resource associated with the error
+        type: string
+        example: 4afb2ce3741fe11d88838cc9
+      message:
+        description: The filled in error message
+        type: string
+        example: >-
+          One or more errors occurred. See the contained list for details of
+          each error.
+      args:
+        description: Positional argument values for the error code
+        type: array
+        items:
+          type: string
+          example: 4afb2ce3741fe11d88838cc9
+      innerErrors:
+        type: array
+        example:
+          - name: TestMonitor.InvalidId
+            code: -253700
+            resourceType: TestResult
+            resourceId: 4afb2ce3741fe11d88838cc9
+            message: Invalid Id.
+            args: []
+        items:
+          $ref: '#/definitions/Error'
+    example:
+      name: Skyline.OneOrMoreErrorsOccurred
+      code: -251040
+      message: >-
+        One or more errors occurred. See the contained list for details of each
+        error.
+      args: []
+      innerErrors:
+        - name: TestMonitor.InvalidId
+          code: -253700
+          resourceType: TestResult
+          resourceId: 4afb2ce3741fe11d88838cc9
+          message: Invalid Id.
+          args: []
+  Operation:
+    description: An operation provided by the API
+    type: object
+    properties:
+      available:
+        type: boolean
+        description: Whether the operation is available to the caller
+      version:
+        type: integer
+        description: The version of the available operation
+    required: [available]
+    example:
+      available: true
+      version: 1
+  V1Operations:
+    description: V1 operations
+    type: object
+    properties:
+      operations:
+        description: >-
+          Available operations in the v1 version of the API:
```

Comment:
This one happens to be odd, because every operation is explicitly identified here.  It may make sense to combine these into more general operations.  Then the explanation of specific requests will make sense.  Since that requires a backend service change, I'll defer to doing that in a separate change.